android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyInterceptor.kt

```diff
@@ -250,4 +261,79 @@ class EnvoyInterceptor : Interceptor {
         urlRequest.start()
         return callback.blockForResponse()
     }
+
+    private fun setupOhttp() {
+        state.ctx?.let {
+
+            // does this benefit from a separate cache?
+            // size?
+            val configRequestsCache: Cache = Cache(
+                directory = File(it.cacheDir, "ohttp"),
+                maxSize = 50L * 1024L * 1024L // 50 MiB
+            )
+
+            // we always use OHTTP for this client
+            val isOhttpEnabled: IsOhttpEnabledProvider = IsOhttpEnabledProvider { true }
+
+            val url = state.activeService!!.url
+            val tempUri = Uri.parse(url)
+
+            // remove query params and convert to HttpUrl
+            // yikes :)
+            val relayUrl = tempUri.buildUpon().clearQuery().build().toString().toHttpUrl()
+            Log.d(TAG, "OHTTP URL: $url")
+
+            tempUri.getQueryParameter("key_url")?.let {
+                val keyUrl = Uri.decode(it).toHttpUrl()
+
+                Log.d(TAG, "OHTTP key URL: $keyUrl")
+
+                val ohttpConfig = OhttpConfig(
+                    relayUrl = relayUrl, // relay server
+                    userAgent = "GreatFire Envoy/Guardian Project OHTTP", // user agent for OHTTP requests to the relay server
```

Comment:
does the server check for this user agent?  if not, i seem to recall at once point we were asked to remove references to "greatfire".

Reply:
It doesn't matter, I just thought it would be polite to identify ourselves to the relay :)

I don't recall being asked to not include it in code, I mean this repo is under GreatFire's account